File: Backend/models/service.py
from enum import Enum
from typing import Optional
from pydantic import BaseModel, EmailStr

class ServiceCategory(str, Enum):
    TECHNOLOGY = 'tecnologia'
    EDUCATION = 'educacion'
    HEALTH = 'salud'
    HOME = 'hogar'
    BUSINESS = 'negocios'
    TRANSPORT = 'transporte'
    CREATIVE = 'creatividad'
    MARKETING = 'marketing'
    OTHER = 'otro'

class ServiceStatus(str, Enum):
    PENDING = "pendiente"
    APPROVED = "aprobado"
    REJECTED = "rechazado"

class Service(BaseModel):
    _id: Optional[str] = None
    category: ServiceCategory = ServiceCategory.OTHER
    name: str
    provider_id: str
    description: str
    price: float
    status: ServiceStatus = ServiceStatus.PENDING
    images: Optional[list[str]]
    
    
    

File: web/src/app/models/service.ts
export interface ServiceModel {
    _id?: string; 
    category: ServiceCategory | string; 
    name: string;
    provider_id?: string;
    description: string;
    price: number;
    status?: ServiceStatus;
    images?: string[]
    provider_name?: string
    provider_email?: string
}

export enum ServiceStatus {
    ALL = 'all',
    PENDING = 'pendiente',
    APPROVED = 'aprovado',
    REJECTED = 'rechazado'
}

export enum ServiceCategory {
    ALL = 'all',
    TECHNOLOGY = 'tecnologia',
    EDUCATION = 'educacion',
    HEALTH = 'salud',
    HOME = 'hogar',
    BUSINESS = 'negocios',
    TRANSPORT = 'transporte',
    CREATIVE = 'creatividad',
    MARKETING = 'marketing',
    OTHER = 'otro',
}